pkg/api/validation/validation.go

```diff
@@ -145,6 +152,23 @@ func ValidateEndpointsSpecificAnnotations(annotations map[string]string, fldPath
 	return allErrs
 }
 
+func validateOwnerReference(ownerReference api.OwnerReference, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	gvk := unversioned.FromAPIVersionAndKind(ownerReference.APIVersion, ownerReference.Kind)
```

Comment:
In offline chat with lavalamp, we decided that **GC should ignore bogus ownerReference**, because in the following two situations, we don't want GC to delete an object: 1. the cluster stops supporting an old version of a resource, we don't want GC to delete objects that have ownerReferences pointing to an object of the old version. 2. if user accidentally misspell an ownerReference, we don't want GC to delete the object.